### ISSUE TITLE
Adding Drag objects to a screen.

### ISSUE THREAD

**Issue by XelaPy** [renpy/renpy#1250]
```renpy
image some_image:
    Null()
    linear .1 xpos 0
    repeat

screen debug():
    # add "some_image"

    draggroup:
        drag:
            droppable True

        add Drag(d=Solid("F00", xysize=(80, 200)), draggable=True)
        
label start:
    call screen debug()
```

This crashes the engine. If there is something on the screen that redraws itself as quickly as possible (.0), such as that image (commented out), everything is working perfectly normal (this may help with understanding the cause somehow).

In order to cause a crash, you must attempt dragging the red drag displayable.

**Traceback:**
```
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/script.rpy", line 16, in script
    call screen debug()
  File "renpy/common/000statements.rpy", line 471, in execute_call_screen
    store._return = renpy.call_screen(name, *args, **kwargs)
TypeError: unsupported operand type(s) for +: 'NoneType' and 'NoneType'

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "game/script.rpy", line 16, in script
    call screen debug()
  File "D:\Coding\RenPy\renpy\ast.py", line 1706, in execute
    self.call("execute")
  File "D:\Coding\RenPy\renpy\ast.py", line 1724, in call
    return renpy.statements.call(method, parsed, *args, **kwargs)
  File "D:\Coding\RenPy\renpy\statements.py", line 145, in call
    return method(parsed, *args, **kwargs)
  File "renpy/common/000statements.rpy", line 471, in execute_call_screen
    store._return = renpy.call_screen(name, *args, **kwargs)
  File "D:\Coding\RenPy\renpy\exports.py", line 2526, in call_screen
    rv = renpy.ui.interact(mouse="screen", type="screen", roll_forward=roll_forward)
  File "D:\Coding\RenPy\renpy\ui.py", line 285, in interact
    rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
  File "D:\Coding\RenPy\renpy\display\core.py", line 2526, in interact
    repeat, rv = self.interact_core(preloads=preloads, trans_pause=trans_pause, **kwargs)
  File "D:\Coding\RenPy\renpy\display\core.py", line 3204, in interact_core
    rv = root_widget.event(ev, x, y, 0)
  File "D:\Coding\RenPy\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "D:\Coding\RenPy\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "D:\Coding\RenPy\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "D:\Coding\RenPy\renpy\display\screen.py", line 651, in event
    rv = self.child.event(ev, x, y, st)
  File "D:\Coding\RenPy\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "D:\Coding\RenPy\renpy\display\dragdrop.py", line 720, in event
    return super(DragGroup, self).event(ev, x, y, st)
  File "D:\Coding\RenPy\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "D:\Coding\RenPy\renpy\display\dragdrop.py", line 584, in event
    drop = self.drag_group.get_best_drop(joined)
  File "D:\Coding\RenPy\renpy\display\dragdrop.py", line 778, in get_best_drop
    overlap = rect_overlap_area(r1, r2)
  File "D:\Coding\RenPy\renpy\display\dragdrop.py", line 821, in rect_overlap_area
    minright = min(x1 + w1, x2 + w2)
TypeError: unsupported operand type(s) for +: 'NoneType' and 'NoneType'

Windows-8-6.2.9200
Ren'Py 6.99.12.4.2187
Testing New GUI 1.0
```

**Comment by renpytom** [comment]
I checked in a semi-fix here, but this is mostly a programming error on your part. 

Basically, when you do:

```
        add Drag(d=Solid("F00", xysize=(80, 200)), draggable=True)
```
you're creating a new drag each time the screen is updated. (And hence, removing the old one.) That's causing these problems. 

you could do:

```
default mydrag =  Drag(d=Solid("F00", xysize=(80, 200)), draggable=True)
add mydrag
```

but it probably just makes sense to use the drag statement, which handles this for you.

**Comment by XelaPy** [comment]
It's kind of the opposite, guess that I went a bit overboard when trying to reduce code to it's bare minimum :)

In the first version of our setup, "drag" screen language statements were used exclusively. We have other projects with more logically demanding Drag And Drag systems using nothing but screen language to create all of the drags, work amazingly well. In this case, it did not work. Issues were numerous, if a player were to call up game menu while dragging, drag would get stuck in it's position, drags would copy themselves in rare cases, not reflecting underlying logic. Saving and Loading was not reliable.

So I decided to try and create all draggable drags using Python before screen is rendered and place those drags in draggroup using the add statement. **All of the issues were fixed! All of them, even saving and loading became 100% reliable!**

But we also show a Sprite Doll (CDD) using "show" statement (master layer) while the drag and drop screen is shown using "show screen" statement (screen layer). That CDD is redrawn at .0 in developer mode and at .1 without it. Took hours to trace down the cause (after config.developer was set to False) and after I have, I just blamed it on CDD. Later however, the whole issue just disappeared even with Doll redrawn at .1... and I became obsessed tracking it down :)

=================================>>>
The code in the front post is the minimum I could track it down to. New check that you've added, does prevent Ren'Py from crashing, but drag also becomes impossible to drop... at least until interaction is restarted I guess. In the video, you see that it is not possible to drop the drag in it's slot, until Game Menu is shown and dismissed:

https://youtu.be/swllt_YIZ6s

We are now using a screen that has an animation on endless repeat as part of games interface! So this issue is completely fixed in our game. _There is no reason to make this a priority_ (at least on our account).
